plugins/package-managers/node/src/main/kotlin/pnpm/ModuleInfo.kt

```diff
@@ -19,12 +19,20 @@
 
 package org.ossreviewtoolkit.plugins.packagemanagers.node.pnpm
 
+import java.io.ByteArrayInputStream
+
 import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.DecodeSequenceMode
 import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.decodeToSequence
 
 private val JSON = Json { ignoreUnknownKeys = true }
 
-internal fun parsePnpmList(json: String): List<ModuleInfo> = JSON.decodeFromString(json)
+internal fun parsePnpmList(json: String): List<ModuleInfo> =
+    JSON.decodeToSequence<List<ModuleInfo>>(
+        ByteArrayInputStream(json.toByteArray()),
+        DecodeSequenceMode.WHITESPACE_SEPARATED
+    ).toList().flatten()
```

Comment:
I believe we should consider to just call `flatten()` here and change the function to return a `Sequence` (which requires to also change the return type for `listModules()` and trivially adjust some tests).

Reply:
As discussed, the function now returns a sequence.